src/flag_gems/runtime/tune_config/config.py

```diff
@@ -0,0 +1,106 @@
+import triton
+
+from .. import backend, device
+
+
+class Config:
+    def __init__(self):
+        self.config = self.get_vendor_tune_config()
+        self.gen_key = "gen"
+        self.loaded_config = {}
+        self.triton_config_default = {"num_stages": 2, "num_warps": 4, "num_ctas": 1}
+        self.load_all()
+
+    def load_all(self):
+        for key in self.config:
+            self.loaded_config[key] = self.get_op_tune_config(key)
+
+    def get_vendor_tune_config(self):
+        return backend.get_tune_config(device.device_instance.vendor_name)
+
+    def _gen_impl(
+        self,
+        gen_config,
+        param_config,
+        iteration_keys,
+        current_config,
+        current_iteration,
+    ):
+        if current_iteration == len(iteration_keys):
+            return [
+                triton.Config(
+                    current_config["META"],
+                    num_warps=current_config["num_warps"],
+                    num_stages=current_config["num_stages"],
+                    num_ctas=current_config["num_ctas"],
+                )
+            ]
+        current_config_list = []
+        param_key = iteration_keys[current_iteration]
+        if param_key in param_config["META"]:
+            config_var_key = param_config["META"][param_key]
+        else:
+            config_var_key = param_config[param_key]
+        if isinstance(config_var_key, int):
+            key_config = [config_var_key]
+        else:
+            key_config = gen_config[config_var_key]
+        for single_value in key_config:
+            if param_key in param_config["META"]:
+                current_config["META"][param_key] = single_value
+            else:
+                current_config[param_key] = single_value
+            config_item = self._gen_impl(
```

Comment:
I suggest using loop instead of recursion.

Reply:
Get.